lib/elixir/lib/list.ex

```diff
@@ -577,6 +577,51 @@ defmodule List do
   end
 
   @doc """
+  If `enum` starts with `prefix`, return true. Otherwise, return false.
+  Note that `prefix` must be a list; only `enum` is able to be an arbitrary enumerable.
+
+  ### Examples
+
+      iex> List.starts_with?([1, 2, 3], [1, 2])
+      true
+      iex> List.starts_with?([:alpha], [])
+      true
+      iex> List.starts_with?([], [:alpha])
+      false
+  """
+  @spec starts_with?(Enum.t, list) :: boolean
+  def starts_with?(enum, prefix) when is_list(enum) do
```

Comment:
This implementation still operates on enumerables but stuff in the `List` module operates on lists. I think you want something like just

```elixir
def starts_with?(list, prefix) when is_list(list) and is_list(prefix) do
  :lists.prefix(prefix, list)
end
```